lib/pharos/phases/configure_etcd.rb

```diff
@@ -0,0 +1,61 @@
+# frozen_string_literal: true
+
+module Pharos
+  module Phases
+    class ConfigureEtcd < Pharos::Phase
+      title 'Configure etcd'
+      CA_PATH = '/etc/pharos/pki'
+
+      register_component(
+        Pharos::Phases::Component.new(
+          name: 'etcd', version: Pharos::ETCD_VERSION, license: 'Apache License 2.0'
+        )
+      )
+
+      def call
+        sync_ca
+
+        logger.info(@host.address) { 'Configuring etcd certs ...' }
+        peer_index = @config.etcd_hosts.find_index { |h| h == @host }
+        exec_script(
+          'configure-etcd-certs.sh',
+          PEER_IP: @host.private_address || @host.address,
+          PEER_NAME: "etcd#{peer_index + 1}",
+          ARCH: @host.cpu_arch.name
+        )
+
+        logger.info(@host.address) { 'Configuring etcd ...' }
+        exec_script(
+          'configure-etcd.sh',
+          PEER_IP: @host.private_address || @host.address,
+          INITIAL_CLUSTER: initial_cluster.join(','),
+          ETCD_VERSION: Pharos::ETCD_VERSION,
+          KUBE_VERSION: Pharos::KUBE_VERSION,
+          ARCH: @host.cpu_arch.name,
+          PEER_NAME: "etcd#{peer_index + 1}"
+        )
+      end
+
+      # @return [Array<String>]
+      def initial_cluster
+        i = 0
+        @config.etcd_hosts.map { |h|
+          i += 1
+          "etcd#{i}=https://#{h.peer_address}:2380"
+        }
+      end
+
+      def sync_ca
+        return if @ssh.file(File.join(CA_PATH, 'ca.pem')).exist?
```

Comment:
Good enough but not ideal: if you upload `ca.pem` first but uploading `ca-key.pem` fails for whatever reason, then re-running `pharos-cluster up` is not going to upload the missing `ca-key.pem`. It would be better to attempt to make these things idempotent and robust against random errors.

Reply:
Should we then just write these files always? I'm pretty sure that there is no speed difference if we check every file vs write every file?

Reply:
Not entirely sure, but eventually we'd need to also be able to update these files, for e.g. cert rotation (5 year validity currently)... I figure this is probably good enough to merge for now, we can later solve this for other files like the `/etc/systemd/system/kubelet.service.d/5-pharos.conf` as well.

Maybe change this to check the `file.exist?` for each cert file if you want to be safe.

Reply:
Well there is the controversial `@ssh.file.with_existing do |file|` :)

Reply:
> Well there is the controversial @ssh.file('/tmp/foofoo').with_existing do |file| :)

Not sure what you mean, this would need `with_not_existing`.

But it needs something much more complex than that if you also want to deal with the file changing... if you replace the CA certs, then you'd also need to restart the daemons using them.